app/src/main/java/com/android/unio/ui/settings/Settings.kt

```diff
@@ -52,35 +62,76 @@ fun SettingsScreen(navigationAction: NavigationAction) {
       }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun SettingsContainer() {
   val context = LocalContext.current
+  val preferences by LocalPreferenceFlow.current.collectAsState()
+
+  /** Location Permissions * */
+  val locationPermissions =
+      rememberMultiplePermissionsState(
+          permissions =
+              listOf(
+                  Manifest.permission.ACCESS_FINE_LOCATION,
+                  Manifest.permission.ACCESS_COARSE_LOCATION))
+  val requestPermissionLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) {
+          permissions ->
+        locationPermissions.launchMultiplePermissionRequest()
+      }
+
+  /** Language * */
+  val language = preferences.get<String>(AppPreferences.LANGUAGE) ?: AppPreferences.Language.default
+  val locale = Locale(language)
+  Locale.setDefault(locale)
+
+  val configuration = context.resources.configuration
+  configuration.setLocale(Locale(language))
+  configuration.setLayoutDirection(locale)
+
+  context.createConfigurationContext(configuration)
+  context.resources.updateConfiguration(configuration, context.resources.displayMetrics)
 
   ProvidePreferenceLocals(flow = LocalPreferenceFlow.current) {
     LazyColumn(
         modifier = Modifier.testTag("SettingsContainer"),
     ) {
       listPreference(
```

Comment:
Consider adding the link to the documentation of the library you use in the comment since it is a somewhat custom solution